### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by ulises2607**
- Add models validations
- Add models tests

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      post2 = Post.create(user:, title: 'Post 2', text: 'Content 2', created_at: 2.days.ago)
+      post3 = Post.create(user:, title: 'Post 3', text: 'Content 3', created_at: 3.days.ago)
+      Post.create(user:, title: 'Post 4', text: 'Content 4', created_at: 4.days.ago)
+
+      last_posts = user.last_three_posts
+
+      expect(last_posts).to eq([post1, post2, post3])
+      expect(last_posts.length).to eq(3)
+    end
+
```
Comment (tufoinnkuo10) [inline spec/models/user_spec.rb]: - [x] nice work. please kindly ensure  all custom methods in the models should have a corresponding unit test. the user model should be tested for,
- presence of name
- if PostsCounter is an integer greater than or equal to zero.
please ensure all your test are passing.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    recent_comments = post.five_recent_comments
+
+    expect(recent_comments).to eq([comment6, comment5, comment4, comment3, comment2])
+    expect(recent_comments.length).to eq(5)
+  end
+
+  it 'updates user post_counter after create' do
+    user = User.create(name: 'Alice', photo: 'https://unsplash.com/photos/AlicePhoto', bio: 'Bio for Alice')
+    Post.create(title: 'Test Post', text: 'This is a test post', user:)
+
```
Comment (tufoinnkuo10) [inline spec/models/post_spec.rb:87]: - [x] please remember all methods should have test. kindly add test to show that.. The comments and likes counter are integers and are greater or equal to zero. kinly ensure all your test are passing. nice work.

File: app/models/post.rb [hunk 3]
```diff
@@ -3,7 +3,10 @@ class Post < ActiveRecord::Base
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
 
-  after_create :update_post_counter # Cambiado a after_create
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :text, presence: true, length: { maximum: 5000 }
+
```
Comment (tufoinnkuo10) [inline app/models/post.rb:10]: - [x] Great job so far. please as stated in the guidlines and for your app to work properly, kindly add validations a requested.
- please ensure the CommentsCounter shouldbe an integer greater than or equal to zero.
- Also, the LikesCounter should be an integer greater than or equal to zero.

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @ulises2607 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/models/post.rb (hunk 3 above)
Comment (ulises2607) [reply app/models/post.rb:10]: Thank you!! 🙌🙌

**Review by Deepakdanger — APPROVED** [review]
Hi @ulises2607, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**